src/town_cmd.cpp

```diff
@@ -207,9 +207,7 @@ void Town::InitializeLayout(TownLayout layout)
 
 void Town::FillCachedName() const
 {
-	char buf[MAX_LENGTH_TOWN_NAME_CHARS * MAX_CHAR_LENGTH];
-	char *end = GetTownName(buf, this, lastof(buf));
-	this->cached_name.assign(buf, end);
+	this->cached_name.assign(GetTownName(this));
```

Comment:
Why assign, rather than the standard = ?

Reply:
```suggestion
	this->cached_name = GetTownName(this);
```
Probably because I made some `TextBuf` changes before, which uses `Assign` so I didn't really think about it?